Week3/db-week-3.js

```diff
@@ -0,0 +1,216 @@
+});
+
+app.post('/submit', async function(req, res) {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const todoQuery = `INSERT INTO todos VALUES ('${req.body.todoid}', '${req.body.reminder}')`;
+    const newList = await execQuery(todoQuery);
+    if (newList.length === 0) {
+      res.render('index', {
+        message: 'A list name or a reminder cannot be empty. Please go back and fill the inputs',
+      });
+    } else {
+      res.render('index', { message: 'Thank you for your patience! Your list is saved!' });
+      connection.end();
+    }
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.get('/addItems', async function(req, res) {
+  try {
+    res.sendFile(__dirname + '/pages/addItems.html');
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.post('/saveAddItem', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const itemQuery = `INSERT INTO items VALUES ('${req.body.task}', '${req.body.isdone}', '${req.body.listName}')`;
+    await execQuery(itemQuery);
+    res.render('index', {
+      message: 'Thank you! Your item is saved to your list!',
+      prevPage: 'Home Page',
+    });
+    connection.end();
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.get('/view', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const listQuery = `select todoid, reminder, item, isdone from todos, items where todos.todoid = items.belongs_to;`;
+    const data = await execQuery(listQuery);
+    res.render('todo', { title: 'Lists', items: data });
+    connection.end();
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.get('/viewOnlyLists', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const onlyListQuery = `select todoid from todos;`;
+    await execQuery(onlyListQuery).then(data => {
+      if (data.length == 0) {
+        res.render('index', {
+          title: 'Only List',
+          message: "You don't have any lists, please create some.",
+        });
+      } else {
+        res.render('todoList', { title: 'Lists', items: data });
+      }
+    });
+    connection.end();
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.put('/update/:item/:isdone', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const updateQuery = `UPDATE items SET isdone = '${req.params.item}' Where items.item = '${req.params.isdone}';`;

```

Comment:
When we develop APIs, we always need to return proper HTTP Codes so that the person / program using our API can understand what is happening. I mean something like this:

```
if (!${req.params.item})
       return res.status(404).send('ID not Found!');
```

Here's more info on the HTTP Codes https://www.restapitutorial.com/httpstatuscodes.html

Could you please add them?